pallets/polkadex-ido/src/lib.rs

```diff
@@ -194,6 +214,29 @@ pub mod pallet {
     #[pallet::call]
     impl<T: Config> Pallet<T> {
 
+        /// Registers a new investor to allow participating in funding round.
+		///
+		/// # Parameters
+		///
+		/// * `origin`: Account to be registered as Investor
+		#[pallet::weight((10_000, DispatchClass::Normal))]
+		pub fn register_investor(origin: OriginFor<T>) -> DispatchResult {
+			let current_block_no = <frame_system::Pallet<T>>::block_number();
+			let who: T::AccountId = ensure_signed(origin)?;
+			ensure!(!<InfoInvestor<T>>::contains_key(&who), Error::<T>::InvestorAlreadyRegistered);
+			let amount: BalanceOf<T> = T::IDOPDXAmount::get();
+			ensure!(
+				T::Currency::reserve(&who, amount).is_ok(),
+				Error::<T>::FailedToMoveBalanceToReserve
+			);
+            // TODO: We are reserving for default 100 blocks, This must be discussed
```

Comment:
There is no time limit for reserving for IDO; it will be reserved until the Investor deregisters.